packages/hub/src/components/layout/UserControls.tsx

```diff
@@ -22,28 +24,40 @@ export function UserControls({ session }: { session: Session | null }) {
     // https://github.com/vercel/next.js/issues/42556 (it's closed but not really solved)
     window.location.href = chooseUsernameRoute();
   }
+  const router = useRouter();
 
   return !session?.user ? (
     <Button onClick={() => signIn()}>Sign In</Button>
   ) : (
     <div className="flex items-center gap-2">
-      {session.user.username === undefined ? null : (
-        <div>
-          <UsernameLink username={session.user.username} />
-        </div>
-      )}
-
-      <DotsDropdown>
-        {() => (
+      <Dropdown
+        render={() => (
           <DropdownMenu>
+            <DropdownMenuHeader>User Actions</DropdownMenuHeader>
+            {!!session.user.username && (
+              <DropdownMenuActionItem
+                onClick={() =>
+                  router.push(
+                    userRoute({ username: session.user.username || "" })
```

Comment:
(minor musing on coding style and type safety)

That `|| ""` scares me, even though it'll never happen.

If I was doing this refactoring, I'd:
- do early return for "not signed in" state
- extract `const { username } = session.user;`
- use `username` in the following code

Then Typescript would guarantee type safety and `|| ""` check won't be necessary.

Reply:
One annoying thing here is that username is an optional property of user. We do already a check for user.

Can we make username a required property? I assume all users should have usernames.

Reply:
Well, usernames are not set initially, and `<UserControls>` must render somehow on `/settings/choose-username`.

We can't set usernames to an empty string on the backend, because username field is unique in Postgres.

We could lie about it in `src/types.d.ts`, but that won't be typesafe.

It would be good to ask next-auth to postpone the creation of a user object in the database until the username is set, but I'm not sure if that's possible.

Reply:
Ah, that makes sense then. In that case, it's probably best to keep. 

Another thing we could hypothetically do is not return a user object until it has a name, but that seems more confusing. (Arguably, it's not really a user until it has a name and stuff)